components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultDomainObjectFactoryTest.java

```diff
@@ -0,0 +1,254 @@
+import org.phenotips.data.rest.DomainObjectFactory;
+import org.phenotips.data.rest.PatientResource;
+import org.phenotips.data.rest.model.Alternatives;
+import org.phenotips.data.rest.model.PatientSummary;
+import org.xwiki.bridge.DocumentAccessBridge;
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.util.DefaultParameterizedType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.DocumentReferenceResolver;
+import org.xwiki.security.authorization.AuthorizationManager;
+import org.xwiki.security.authorization.Right;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+import org.xwiki.users.UserManager;
+
+import javax.inject.Named;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import java.lang.reflect.ParameterizedType;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class DefaultDomainObjectFactoryTest {
+
+    @Rule
+    public MockitoComponentMockingRule<DomainObjectFactory> mocker =
+            new MockitoComponentMockingRule<DomainObjectFactory>(DefaultDomainObjectFactory.class);
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private UriInfo uriInfo;
+
+    private ParameterizedType stringResolverType = new DefaultParameterizedType(null, DocumentReferenceResolver.class,
+            String.class);
+
+    private AuthorizationManager access;
+
+    private UserManager users;
+
+    private DocumentAccessBridge documentAccessBridge;
+
+    @Named("current")
+    private DocumentReferenceResolver<String> stringResolver;
+
+    @Before
+    public void setUp() throws ComponentLookupException {
+        MockitoAnnotations.initMocks(this);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+        this.documentAccessBridge = this.mocker.getInstance(DocumentAccessBridge.class);
+        this.stringResolver = this.mocker.getInstance(this.stringResolverType, "current");
+
+        when(this.users.getCurrentUser()).thenReturn(null);
+    }
+
+    @Test
+    public void cannotCreatePatientWithNoAccess() throws ComponentLookupException {
+        when(this.patient.getDocument()).thenReturn(null);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(false);
+        assertNull(this.mocker.getComponentUnderTest().createPatientSummary(this.patient, this.uriInfo));
+    }
+
+    @Test
+    public void createPatientDocumentExceptionReturnsNull() throws Exception {
+        DocumentReference documentReference = mock(DocumentReference.class);
+        when(this.patient.getDocument()).thenReturn(documentReference);
+        when(this.access.hasAccess(Right.VIEW, null, null)).thenReturn(true);
```

Comment:
this should be changed to `when(this.access.hasAccess(Right.VIEW, null, documentReference)).thenReturn(true);` else you are not effectively setting the behaviour of `hasAccess()` for this test since `patient.getDocument()` will return a mock object, not null